### PR TITLE
Deduplicate kubectl functions and simplify retrying applies

### PR DESCRIPTION
**Pull request by markusthoemmes**
# Changes

As the title says really:

1. Extracted `waitForXXX` functions to deduplicate super long kubectl commands.
2. Extracted `retryingApply` to consistently retry on all of the remote apply applications. I didn't quite parse why the `wait` package was used to be honest as it seems like we'd want the applys to be retried a few times. This has the nice sideeffect of dropping a dependency.

/kind cleanup

/assign @psschwei

### CODE REVIEW

**Review by psschwei — APPROVED** [review]
This is great, thanks !!!

Want to give @julz a chance to weigh in (as he originally suggested using `wait` instead of retries)

/lgtm
/hold

File: pkg/install/install.go [hunk 1]
```diff
@@ -217,3 +177,30 @@ func runCommand(c *exec.Cmd) error {
...
+		err := runCommand(cmd)
+		if err == nil {
+			break
+		}
+		if i >= 3 {
+			return err
+		}
+		time.Sleep(10 * time.Second)
+	}
+	return nil
```
Comment (julz) [inline pkg/install/install.go:195]: can this be:

```suggestion
	var err error
	for i := 0; i <= 3 ; i++ {
		if err = runCommand(cmd); err == nil {
			break
		}
		
		time.Sleep(10 * time.Second)
	}
	return err
```

**Review by julz — COMMENTED** [review]
much nicer - lgtm apart from some potential code golf 🤓

File: pkg/install/install.go (hunk 1 above)
Comment (markusthoemmes) [reply pkg/install/install.go]: Done (with a slight change). I personally prefer to have assignments to `err` outside of the `if v == x(); err == nil` construct, as that's typically used with a local-to-the-if block error variable.

Comment (julz) [reply pkg/install/install.go]: heh, I actually almost suggested it that way for the same reason - wfm

**Review by julz — APPROVED** [review]
🏌️ 

/lgtm
/approve

**Pull request merged**